icepyx/tests/test_quest_argo.py

```diff
@@ -0,0 +1,233 @@
+import pytest
+import re
+
+from icepyx.quest.quest import Quest
+
+# create an Argo instance via quest (Argo is a submodule)
+@pytest.fixture(scope="function")
+def argo_quest_instance():
+    def _argo_quest_instance(bounding_box, date_range):  # aka "factories as fixtures"
+        my_quest = Quest(spatial_extent=bounding_box, date_range=date_range)
+        my_quest.add_argo()
+        my_argo = my_quest.datasets["argo"]
+
+        return my_argo
+
+    return _argo_quest_instance
+
+
+# ---------------------------------------------------
+# Test Formatting and Validation
+
+
+def test_fmt_coordinates(argo_quest_instance):
+    reg_a = argo_quest_instance([-154, 30, -143, 37], ["2022-04-12", "2022-04-26"])
+    obs = reg_a._fmt_coordinates()
+
+    exp = "[[-143.0,30.0],[-143.0,37.0],[-154.0,37.0],[-154.0,30.0],[-143.0,30.0]]"
+
+    assert obs == exp
+
+
+def test_validate_parameters(argo_quest_instance):
+    reg_a = argo_quest_instance([-154, 30, -143, 37], ["2022-04-12", "2022-04-26"])
+
+    invalid_params = ["temp", "temperature_files"]
+
+    ermsg = re.escape(
+        "Parameter '{0}' is not valid. Valid parameters are {1}".format(
+            "temp", reg_a._valid_params()
+        )
+    )
+
+    with pytest.raises(AssertionError, match=ermsg):
+        reg_a._validate_parameters(invalid_params)
+
+
+# ---------------------------------------------------
+# Test Setters
+
+
+def test_param_setter(argo_quest_instance):
+    reg_a = argo_quest_instance([-154, 30, -143, 37], ["2022-04-12", "2022-04-26"])
+
+    exp = ["temperature"]
+    assert reg_a.params == exp
+
+    reg_a.params = ["temperature", "salinity"]
+
+    exp = list(set(["temperature", "salinity"]))
+    assert reg_a.params == exp
+
+
+def test_param_setter_invalid_inputs(argo_quest_instance):
+    reg_a = argo_quest_instance([-154, 30, -143, 37], ["2022-04-12", "2022-04-26"])
+
+    exp = ["temperature"]
+    assert reg_a.params == exp
+
+    ermsg = re.escape(
+        "Parameter '{0}' is not valid. Valid parameters are {1}".format(
+            "temp", reg_a._valid_params()
+        )
+    )
+
+    with pytest.raises(AssertionError, match=ermsg):
+        reg_a.params = ["temp", "salinity"]
+
+
+def test_presRange_setter(argo_quest_instance):
+    reg_a = argo_quest_instance([-154, 30, -143, 37], ["2022-04-12", "2022-04-26"])
+
+    exp = None
+    assert reg_a.presRange == exp
+
+    reg_a.presRange = "0.5,150"
+
+    exp = "0.5,150"
+    assert reg_a.presRange == exp
+
+
+def test_presRange_setter_invalid_inputs(argo_quest_instance):
+    reg_a = argo_quest_instance([-154, 30, -143, 37], ["2022-04-12", "2022-04-26"])
+
+    exp = None
+    assert reg_a._presRange == exp
+
+    reg_a._presRange = (
```

Comment:
@RomiP Can you explain making this private to me? In my mind this should make the test fail because we never try to "set" reg_a.params (as a user would) with the invalid pressure range, so we wouldn't get an error message if we submitted it. But it DOES pass, so I'm clearly not getting something. I also don't see how "setting" _presRange effectively tests the setter, since the setter is on `presRange` not `_presRange`...

Reply:
Follow up: undid changes given that there are getter/setter methods to access the private variable